### ISSUE TITLE
Incompatible with lita-slack?

### ISSUE THREAD

**Issue by cypgrumpy** [dfinninger/lita-cmd#11]
Works perfectly when the bot is using the :shell handler, but when using :slack sending cmd list to the bot does nothing.

**Comment by dfinninger** [comment]
Hm, that kinda defeats the purpose now, doesn't it?
I'll take a look.

**Comment by dfinninger** [comment]
Found the issue here:
https://github.com/dfinninger/lita-cmd/blob/156a04086a2c4f2e8ba5a5fd3c6b34727db73f91/lib/lita/handlers/cmd.rb#L25

This line was added in to address an issue when the `command_prefix` was empty. However, when I play with it, both the `:shell` and `:slack` adapters run fine without the line.

@DannyBen What are your thoughts on removing this line?

**Comment by DannyBen** [comment]
Well, if we remove it we practically kill the option for an empty command prefix no? I admit, this line of code is quite patchy and the empty command prefix does introduce some uncomfortable hacks, but its a nice feature no?

What is the use case that causes an issue?

If I understand correctly, and you saw it working fine without the line in both shell and slack, it will probably fail in hipchat.... this is why I added it if memory serves.

**Comment by dfinninger** [comment]
It is a nice feature, in that it does allow for adding lita "handlers" without needing to do much configuration at all. However, it does collide with the built-in lita commands:

```
Lita > lita help
[2015-12-31 17:48:51 UTC] DEBUG: Dispatching message to Lita::Handlers::Help#help.
lita: help - Lists help information for terms and command the robot will respond to.
lita: help COMMAND - Lists help information for terms or commands that begin with COMMAND.
lita: info - Replies with the current version of Lita.
lita: users find SEARCH_TERM - Find a Lita user by ID, name, or mention name.
lita: ACTION - run the specified ACTION. use `Lita list` for a list of available actions.
[2015-12-31 17:48:51 UTC] DEBUG: Dispatching message to Lita::Handlers::Cmd#run_action.
Unauthorized to run 'help'!    # <= lita-cmd taking the command after lita-help
```

But this regex breaks Slack integration. Which really kills this plugin for a good deal of people. I don't know about Hipchat because I don't have an account, but I can get one to do more testing.

The regex _shouldn't_ match against the `script` variable, but it is for some reason. Only when using the slack handler does it return on that line. I don't know what the difference is yet. I can look at it more later this afternoon, but I'd rather not break the plugin for a nice feature if it comes down to it.

**Comment by DannyBen** [comment]
Of course, if it comes to working on slack vs. empty prefix, it is clear who needs to win. I believe we can easily support both. I have both slack and hipchat accounts, I will check hopefully tonight and write back. If you want to release a version without this line until we fix it, its your call.

@cypgrumpy - how did you name your bot? I mean, when you execute a command, what exactly do you write? private / public message?

EDIT:
As I look at the code I see no reason for it NOT to work, especially if the command prefix is not empty. 

If `command_prefix` is set to "run " then "botmane run list" should return list of commands, "botname help" will show the normal bot help. The line you marked as the offending line works as it should, I dont see how it can block a call for "list" or "help" (unless the botname is list or help...)

**Comment by dfinninger** [comment]
I'll be testing  on my end to try and figure it out as well. I think it'll
be fine for the next couple days to try to figure out what is going wrong
before needing to push out a patch.

**Comment by DannyBen** [comment]
Cool. I am testing as well, and it is reproducible at least.

**Comment by DannyBen** [comment]
Found the problem. The `robot.name` was not set by slack (so the regex matched against empty string) . Not sure if it happened in hipchat or not, but in any case I made a simple fix to use `robot.mention_name` if available. Will post a PR soon.

**Comment by dfinninger** [comment]
Great!

**Comment by dfinninger** [comment]
Fixed by #12
